server/channels/store/sqlstore/thread_store.go

```diff
@@ -1137,3 +1137,25 @@ func (s *SqlThreadStore) updateThreadParticipantsForUserTx(trx *sqlxTxWrapper, p
 
 	return nil
 }
+
+// UpdateTeamIdForChannelThreads updates the team id for all threads in a channel.
+// Specifically used when a channel is moved to a different team.
+// If a user is not member of the new team, the threads will be deleted by the
+// channel move process.
+func (s *SqlThreadStore) UpdateTeamIdForChannelThreads(channelId, teamId string) error {
+	query := s.getQueryBuilder().
+		Update("Threads").
+		Set("ThreadTeamId", teamId).
+		Where(
+			sq.And{
+				sq.Eq{"ChannelId": channelId},
+				sq.Expr("EXISTS(SELECT 1 FROM Teams WHERE Id = ?)", teamId),
```

Comment:
For my own interest, can you explain this exists clause? Is it to ensure that `teamId` is valid?

Reply:
Indeed, because there is no other foreign keys here, it's best to have a sanity check here.